pkg/plugins/trivy/image.go

```diff
@@ -115,52 +110,62 @@ func GetPodSpecForStandaloneMode(ctx trivyoperator.PluginContext,
 			},
 		},
 	}
-
 	if volume, volumeMount := config.GenerateSslCertDirVolumeIfAvailable(trivyConfigName); volume != nil && volumeMount != nil {
 		volumes = append(volumes, *volume)
 		volumeMounts = append(volumeMounts, *volumeMount)
 	}
 
 	var initContainers []corev1.Container
 
+	cacheDir := config.GetImageScanCacheDir()
+
 	args := []string{
 		"--cache-dir",
 		cacheDir,
 		"image",
-		"--download-db-only",
-		"--db-repository",
-		dbRepository,
 	}
 	if config.ConfigFileExists() {
 		args = append(args, "--config", configFileMountPath)
 	}
 
-	initContainers = append(initContainers, corev1.Container{
-		Name:                     p.idGenerator.GenerateID(),
-		Image:                    trivyImageRef,
-		ImagePullPolicy:          corev1.PullPolicy(config.GetImagePullPolicy()),
-		TerminationMessagePolicy: corev1.TerminationMessageFallbackToLogsOnError,
-		Env:                      initContainerEnvVar(trivyConfigName, config),
-		Command: []string{
-			"trivy",
-		},
-		Args:            args,
-		Resources:       requirements,
-		SecurityContext: securityContext,
-		VolumeMounts:    volumeMounts,
-	})
-
-	if !config.GetSkipJavaDBUpdate() && config.TrivyDBRepositoryCredentialsSet() {
-		argsDBUpdater := []string{
-			"--cache-dir",
-			cacheDir,
-			"image",
-			"--download-java-db-only",
-			"--java-db-repository",
-			config.GetJavaDBRepository(),
+	trivyImageRef, err := config.GetImageRef()
+	if err != nil {
+		return corev1.PodSpec{}, nil, err
+	}
+	resourceRequirements, err := config.GetResourceRequirements()
+	if err != nil {
+		return corev1.PodSpec{}, nil, err
+	}
+
+	if config.TrivyDBRepositoryCredentialsSet() {
+		dockerConfigVolumeMount := []corev1.VolumeMount{
+			{
+				Name:      dockerConfigVolumeName,
+				ReadOnly:  false,
+				MountPath: "/root/.docker",
```

Comment:
Just for my own understanding, will this mount point be the same across different environments? Should we dynamically determine this?

Reply:
you're right in common case - this dir can be different for different systems.
but in our case - we run a pod without security context, and it runs in the same docker image (trivy) based on alpine, so "/root/.docker" should be the same.

Reply:
now it uses a specific docker folder instead of a default dir
https://github.com/aquasecurity/trivy-operator/pull/2595/commits/7c6576689feb6ec5ec07b202bcd074b8c222070f